File: pinyougou-web-manager/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		//最终跟踪分析到这里:去写下面相应的后台controller
		return $http.get('../specification/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../specification/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../specification/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		/*
		然后追踪到这里:找到了后台对应的Controller方法名
			以及要传递的参数名称:
			然后去后台写代码
			注意该方法的传参:entity内部结构,去创建一个entity对应的Pojo对象
		 */
		return  $http.post('../specification/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../specification/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../specification/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../specification/search.do?page='+page+"&rows="+rows, searchEntity);
	}  
	
	this.selectOptionList=function(){
		return $http.get("../specification/selectOptionList.do");
	}
});

File: pinyougou-interface/src/main/java/cn/itcast/core/service/SpecificationService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.specification.Specification;
import entity.PageResult;
import entity.Result;
import pojogroup.SpecificationVo;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
//    Search:页面打开首次加载一些数据:
    public PageResult search(Integer page,Integer rows,Specification specification);

//    add:
    Result add(SpecificationVo specificationVo);

    SpecificationVo findOne(Long id);

    Result update(SpecificationVo specificationVo);

    List<Map> selectOptionList();
}
